pyem/star.py

```diff
@@ -190,6 +190,38 @@ def merge_key(s1, s2, threshold=0.5):
     return None
 
 
+def remove_cryosparc_leading_uid(df, inplace=False):
+    """
+    Remove leading UIDs generated by cryoSPARC from path names if they exist.
+
+    For example:
+    Before: J79/imported/002635156430113862662_14sep05c_00024sq_00003hl_00002es_frames.mrc
+    After: J79/imported/14sep05c_00024sq_00003hl_00002es_frames.mrc
+    """
+    df = df if inplace else df.copy()
+    # always at the beginning of the path (given we're using the basename)
+    # 21 numbers, followed by an underscore
+    uid_match_re = r'^\d{21}_'
+    path_columns = [
+        Relion.IMAGE_NAME,
+        UCSF.IMAGE_BASENAME,
+        Relion.MICROGRAPH_NAME,
+        UCSF.MICROGRAPH_BASENAME]
+
+    for column_name in path_columns:
+        if column_name in df:
+            # check if there are any leading UIDs in any of the paths in this column
+            leading_uid_exists = any(filter(lambda path: re.match(
+                uid_match_re, os.path.basename(path)), list(df[column_name])))
+            if leading_uid_exists:
+                # remove the leading UID
+                df[column_name] = df[column_name].apply(
+                    lambda x: os.path.join(os.path.dirname(x), 
+                        '_'.join(os.path.basename(x).split('_')[1:])))
```

Comment:
I just realized this assumes all paths have the UID- maybe I should change this such that it only modifies the ones that do.

Reply:
addressed in 1aab197da75c92d04caae97c0fcd96abd1d987bc